wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditReportPrinter.java

```diff
@@ -0,0 +1,175 @@
+
+import static com.google.common.io.Files.append;
+
+/**
+ * Audit report printer.
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+class AuditReportPrinter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditReportPrinter.class);
+
+    /**
+     * Prints header of audit report in format:
+     * <p>
+     * Number of all users: <value>
+     * Number of users licensed: <value>
+     * Date when license expires: <date>
+     * <p>
+     * If license is not defined it prints error message instead license rows.
+     *
+     * @param auditReport
+     *         file of audit report
+     * @param allUsersNumber
+     *         all users number
+     * @param license
+     *         {@link CodenvyLicense} object
+     * @throws ServerException
+     *         if an error occurs
+     */
+    void printHeader(Path auditReport, long allUsersNumber, @Nullable CodenvyLicense license) throws ServerException {
+        printRow("Number of all users: " + allUsersNumber + "\n", auditReport);
+        if (license != null) {
+            printRow("Number of users licensed: " + license.getNumberOfUsers() + "\n", auditReport);
+            printRow("Date when license expires: " + new SimpleDateFormat("dd MMMM yyyy").format(license.getExpirationDate()) + "\n",
+                     auditReport);
+        } else {
+            printError("Failed to retrieve license", auditReport);
+        }
+    }
+
+    /**
+     * Print users info and his related workspaces info in format:
+     * <p>
+     * <user e-mail> is owner of <own workspaces number> workspace(s) and has permissions in <workspaces number> workspace(s)
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * .
+     * .
+     * .
+     * └ <workspace name>, is owner: <true or false>, permissions: <list of permissions for current user>
+     * <p>
+     * If the list of user workspaces is not defined it prints error message.
+     *
+     * @param auditReport
+     *         file of audit report
+     * @param user
+     *         {@link UserImpl} object
+     * @param workspaces
+     *         list of workspaces that are related to given user
+     * @param wsPermissions
+     *         map of permissions to workspaces
+     * @throws ServerException
+     *         if an error occurs
+     */
+    void printUserInfoWithHisWorkspacesInfo(Path auditReport, UserImpl user, @Nullable List<WorkspaceImpl> workspaces,
+                                            Map<String, List<AbstractPermissions>> wsPermissions)
+            throws ServerException {
+
+        if (workspaces == null) {
+            printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+            return;
+        }
+
+        int workspacesNumber = workspaces.size();
+        long ownWorkspacesNumber = workspaces.stream().filter(workspace -> workspace.getNamespace().equals(user.getName())).count();
```

Comment:
It can be false in case when user is owner of workspace but doesn't have direct permissions